File: src/leetcode/bit/java/NumberOf1Bits.java
package leetcode.bit.java;

public class NumberOf1Bits {
    public static int hammingWeight(int n) {
        int res = 0;
        for (int i = 0; i < 32; i++) {
            res += n & 1;
            System.out.println("res" + res);
            n >>= 1;
        }
        return res;
    }

    public static void main(String[] args) {
        int r = NumberOf1Bits.hammingWeight(9);
        System.out.println(r);
    }
}

File: src/leetcode/bit/kotlin/NumberOf1Bits.kt
package leetcode.bit.kt

fun hammingWeight(n:Int):Int {
    if (n == 0) return 0

    var res = 0
    var num = n
    for(i in 0 until 32) {
        res += num and 1
        num = num shr 1
    }
    return res
}